### PR TITLE
feat(form): add Dropdown control

### LINKED ISSUE #143: feat(form): Dropdown control
We don't have support now for a dropdown 😅

- Implement a new `dropdown` control type with configuration properties similar to the discussed configuration of radio in #142 
- Create a new `Dropdown.astro` control component that will be rendered when `control.type` is dropdown (see implementation for `RadioGroup.astro`
- `Dropdown.astro` will render `<select>` and several `<option>` tags depending on the control  configuration

### PR DESCRIPTION
**Pull request by Icyscools**
feat(form): Add Dropdown Control

Fixes #143 

Description of changes:
- Implement a `Dropdown.astro` control.
- Add `Dropdown` and `DropdownOption` type. 
These types have attributes like `Radio` and `RadioOption` types but I created it for use in case the dropdown has different attributes than that in the future.
- Include tests on a `Dropdown` control.

Tag a reviewer: @ayoayco 

Tasks:
- [x] I have ran the build command to make sure apps are working: `npm run build`
- [x] I have ran the tests to make sure nothing is broken: `npm run test`
- [x] I have ran the linter to make sure code is clean: `npm run lint`
- [x] I have reviewed my own code to remove changes that are not needed

### CODE REVIEW

File: packages/common/types/control.types.ts [hunk 1]
```diff
@@ -53,6 +54,17 @@ export interface RadioOption {
   value: string;
 }
 
+export interface Dropdown extends Omit<ControlBase, "value"> {
+  type: "dropdown";
+  value?: string;
+  options: string[] | DropdownOption[];
+}
+
+export interface DropdownOption {
```
Comment (ayoayco) [inline packages/common/types/control.types.ts:64]: Lets have a `ControlOption` instead that is used by both radio and dropdown :)

File: packages/form/core/form-control.ts [hunk 2]
```diff
@@ -5,16 +5,18 @@ import type {
...
 } from '@astro-reactive/common';
 
-export type ControlConfig = ControlBase | Checkbox | Radio | Submit | Button;
+export type ControlConfig = ControlBase | Checkbox | Radio | Submit | Button | Dropdown;
 
 export class FormControl {
 	private _name = '';
 	private _type: ControlType = 'text';
-	private _value?: string | number | null | string[] | RadioOption[];
+	private _value?: string | number | null | string[] | RadioOption[] | DropdownOption[];
```
Comment (ayoayco) [inline packages/form/core/form-control.ts:19]: Then we can just have `ControlOption[]` here :)

File: packages/form/test/Dropdown.astro.test.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+		// act
+		component = await getComponentOutput('./components/controls/Dropdown.astro', props);
+		const actualResult = cleanString(component.html);
+		const matches = actualResult.match(element) || [];
+
+		// assert
+		expect(matches.length).to.equal(expectedOptions);
+	});
+
+	it('Should render all dropdown DropdownOption[] options', async () => {
```
Comment (ayoayco) [inline packages/form/test/Dropdown.astro.test.js:36]: Do that forget to update tests when you implement ControlOption :)

**Review by ayoayco — CHANGES_REQUESTED** [review]
I see you're really learning the codebase! Thanks! I just have a suggestion to implement ContrilOption instead of having separate RadioOption and DropdownOption :)

File: packages/common/types/control.types.ts (hunk 1 above)
Comment (Icyscools) [reply packages/common/types/control.types.ts:64]: Sure!

**Comment by Icyscools** [comment]
Done! @ayoayco 🚀  

I change `RadioOption` and `DropdownOption` to `ControlOption` as you suggested, also I add `InputType` type for input field since its required `HTMLInputTypeAttribute | null | undefined` which `HTMLInputTypeAttribute` doesn't include a dropdown so I decided to add this.

**Review by ayoayco — APPROVED** [review]
Yeah having that InputType makes sense. Thanks! Approved! 🚀

**Pull request merged**